src/com/amazon/ionelement/api/IonOperators.kt

```diff
@@ -0,0 +1,410 @@
+package com.amazon.ionelement.api
+
+import com.amazon.ion.Decimal
+import com.amazon.ionelement.api.IntElementSize.*
+import java.math.BigDecimal
+import java.math.BigInteger
+
+/*
+ * This file contains operator overloads and associated extension functions for various IonElement subclasses.
+ * Infix operators that mix IonElement types with native types are only defined for IonElement on the left-hand side.
+ * Augmented assignment operators, e.g., '+=' are provided by Kotlin automatically from the associated binary operator.
+ */
+
+@RequiresOptIn(message = "This API is experimental and subject to change.")
+@Retention(AnnotationRetention.BINARY)
+@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION)
+public annotation class IonOperators
+
+/* Type aliases for readability */
+private typealias BE = BoolElement
+private typealias IE = IntElement
+private typealias DE = DecimalElement
+private typealias FE = FloatElement
+
+/* BoolElement unary operators */
+@IonOperators
+public operator fun BE.not(): BE = ionBool(!this.booleanValue, this.annotations, this.metas)
+
+/* BoolElement with Boolean operators */
+@IonOperators
+public infix fun BE.and(other: Boolean): BE = ionBool(this.booleanValue && other, this.annotations, this.metas)
+@IonOperators
+public infix fun BE.or(other: Boolean): BE = ionBool(this.booleanValue || other, this.annotations, this.metas)
+
+/* Boolean with BoolElement operators */
+@IonOperators
+public infix fun Boolean.and(other: BE): BE = ionBool(this && other.booleanValue, other.annotations, other.metas)
+@IonOperators
+public infix fun Boolean.or(other: BE): BE = ionBool(this || other.booleanValue, other.annotations, other.metas)
+
+/* IntElement unary operators (unaryPlus omitted as bigInteger does not support it) */
+@IonOperators
+public operator fun IE.unaryMinus(): IE =
+    if (useLong(this)) {
+        ionInt(-this.longValue, this.annotations, this.metas)
+    } else {
+        ionInt(-this.bigIntegerValue, this.annotations, this.metas)
+    }
+
+@IonOperators
+public operator fun IE.inc(): IE =
+    if (useLong(this)) try {
+        ionInt(Math.incrementExact(this.longValue), this.annotations, this.metas)
+    } catch (e: ArithmeticException) {
+        ionInt(this.bigIntegerValue.inc(), this.annotations, this.metas)
+    }
+    else ionInt(this.bigIntegerValue.inc(), this.annotations, this.metas)
```

Comment:
Instead of using try/catch, couldn't you just check if the value is `Long.MAX_VALUE`? I.e.:
```suggestion
    if (useLong(this) && this.longValue != Long.MAX_VALUE) {
        ionInt(this.longValue++, this.annotations, this.metas)
    } else {
        ionInt(this.bigIntegerValue.inc(), this.annotations, this.metas)
    }
```